gazebo_plugins/src/gazebo_ros_camera.cpp

```diff
@@ -470,6 +460,66 @@ void GazeboRosCamera::Load(gazebo::sensors::SensorPtr _sensor, sdf::ElementPtr _
 
     impl_->cloud_msg_.header.frame_id = impl_->frame_name_;
   }
+
+  // Dynamic reconfigure
+  if (impl_->sensor_type_ == GazeboRosCameraPrivate::CAMERA) {
+    impl_->ros_node_->declare_parameter(
+      "update_rate", gazebo::CameraPlugin::parentSensor->UpdateRate());
+  } else if (impl_->sensor_type_ == GazeboRosCameraPrivate::DEPTH) {
+    impl_->ros_node_->declare_parameter(
+      "update_rate", gazebo::DepthCameraPlugin::parentSensor->UpdateRate());
+  } else {
+    impl_->ros_node_->declare_parameter(
+      "update_rate", MultiCameraPlugin::parent_sensor_->UpdateRate());
+  }
+
+  auto existing_callback = impl_->ros_node_->set_on_parameters_set_callback(nullptr);
+  auto param_change_callback =
+    [this, existing_callback](std::vector<rclcpp::Parameter> parameters) {
+      auto result = rcl_interfaces::msg::SetParametersResult();
+      if (nullptr != existing_callback) {
+        result = existing_callback(parameters);
+        if (!result.successful) {
+          return result;
+        }
+      }
+      for (const auto & parameter : parameters) {
+        if (nullptr != impl_->trigger_sub_) {
+          RCLCPP_WARN(impl_->ros_node_->get_logger(),
+            "Cannot set update rate for triggered camera");
+          result.successful = false;
+        } else {
+          rclcpp::ParameterType parameter_type = parameter.get_type();
+          if (rclcpp::ParameterType::PARAMETER_DOUBLE == parameter_type) {
+            double rate = parameter.as_double();
+
+            if (impl_->sensor_type_ == GazeboRosCameraPrivate::CAMERA) {
+              gazebo::CameraPlugin::parentSensor->SetUpdateRate(rate);
+            } else if (impl_->sensor_type_ == GazeboRosCameraPrivate::DEPTH) {
+              gazebo::DepthCameraPlugin::parentSensor->SetUpdateRate(rate);
+            } else {
+              MultiCameraPlugin::parent_sensor_->SetUpdateRate(rate);
+            }
+            result.successful = true;
```

Comment:
Since we could be setting multiple parameters, should the result be successful only if all parameters were successfully set (i.e. false if at least one param fails)?